### ISSUE TITLE
[eslint-plugin-lit-a11y]: treeitem has normally textContent, but `lit-a11y/role-has-required-aria-attrs` expects it only to have a aria-label, aria-labelledby or title.

### ISSUE THREAD

**Issue by gerjanvangeest** [open-wc/open-wc#2894]
## Expected behavior

A test fixture with a `role="treeitem"` should be considered valid when it has text content.

```html
<li role="treeitem">Document structure</li>
```
Example copied from: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/treeitem_role

## Actual Behavior

The linter fails, as it on checks on `hasAccessibleName` (which checks for aria-label, aria-labelledby or title attribute) and not `hasAccessibleChildren` (which checks if it has a text content).

https://github.com/open-wc/open-wc/blob/aebf96ccf0bf978f5012f6cc4e0a2ea5153aef80/packages/eslint-plugin-lit-a11y/lib/rules/accessible-name.js#L176

**Comment by thepassle** [comment]
Happy to accept a PR for this :)
